### PR TITLE
For #26286 improve homeThreeDotMenuItemsTest UI test coverage and other refactoring work

### LINKED ISSUE #28170: Intermittent UI test failure - < SmokeTest.mainMenuAddToHomeScreenTest>
### Firebase Test Run:
[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5372505926588932212/executions/bs.3e7cf789072eb77e/testcases/6/test-cases)
[Firebase link](https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/5372505926588932212/executions/bs.a161a9cb342b399e/videos)
### Stacktrace: androidx.test.uiautomator.UiObjectNotFoundException: UiSelector[RESOURCE_ID=org.mozilla.fenix.debug:id/toolbar]
	at androidx.test.uiautomator.UiObject.click(UiObject.java:416)
	at org.mozilla.fenix.ui.robots.HomeScreenRobot$Transition.openNavigationToolbar(HomeScreenRobot.kt:579)
	at org.mozilla.fenix.ui.SmokeTest.mainMenuAddToHomeScreenTest(SmokeTest.kt:274)
    
### Build: Failed on #28149

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-22137)

### PR DESCRIPTION
**Pull request by AndiAJ**
- Create new `MatcherHelper` object helper.

- For #26286 improve `homeThreeDotMenuItemsTest` UI test coverage and other refactoring work.

- For #28170 fix failing `mainMenuAddToHomeScreenTest` UI test

homeThreeDotMenuItemsTest

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26286
Fixes #28170

### CODE REVIEW

**Review by AaronMT — APPROVED** [review]
I'm all for code reduction and smarter ways to re-use code. Thanks.

File: app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+
+    fun assertAppItemsStateWithResourceId(vararg appItems: UiObject) {
+        for (appItem in appItems) {
+            assertTrue(appItem.waitForExists(waitingTime))
+        }
+    }
+
+    fun assertAppItemsStateWithResourceIdAndText(vararg appItems: UiObject) {
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt:68]: A small tweak for clarity: assertItemWithResIdAndTextExists

File: app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    fun appItemContainingText(itemText: String) =
+        mDevice.findObject(UiSelector().textContains(itemText))
+
+    fun appItemWithDescription(description: String) =
+        mDevice.findObject(UiSelector().descriptionContains(description))
+
+    fun appItemStateWithResourceId(resourceId: String, state: Boolean) =
+        mDevice.findObject(UiSelector().resourceId(resourceId).checked(state))
+
+    fun appItemStateWithResourceIdAndText(resourceId: String, text: String, state: Boolean) =
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt:30]: I think replacing the "state" parameter with "isChecked: boolean" will add more clarity when using these methods (without looking at their definition). Same for the "state" part in the method name. Could be confused with the "exists" state check below.

Also, a small change, optional: I think you can remove the "app" part from all these matchers names

**Review by sv-ohorvath — APPROVED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt (hunk 2 above)
Comment (sv-ohorvath) [reply app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt:30]: Same, maybe add an if else statement if the element is not checked?

Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt]: > Same, maybe add an if else statement if the element is not checked?

If it's not checked you just pass `false` for `isChecked` from `checkedItemWithResIdAndText()`

**Pull request merged**